app/forms/declaration/TransportInformation.scala

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms.declaration
+
+import forms.MetadataPropertiesConvertable
+import play.api.data.Forms.{boolean, optional, text}
+import play.api.data.{Form, Forms}
+import play.api.libs.json.Json
+import services.Countries.allCountries
+import utils.validators.forms.FieldValidator._
+
+case class TransportInformation(
+  inlandModeOfTransportCode: Option[String],
+  borderModeOfTransportCode: String,
+  meansOfTransportOnDepartureType: String,
+  meansOfTransportOnDepartureIDNumber: Option[String],
+  meansOfTransportCrossingTheBorderType: String,
+  meansOfTransportCrossingTheBorderIDNumber: Option[String],
+  meansOfTransportCrossingTheBorderNationality: Option[String],
+  container: Boolean
+) extends MetadataPropertiesConvertable {
+
+  override def toMetadataProperties(): Map[String, String] =
+    Map(
+      "declaration.goodsShipment.consignment.arrivalTransportMeans.modeCode" -> inlandModeOfTransportCode.getOrElse(""),
+      "declaration.borderTransportMeans.modeCode" -> borderModeOfTransportCode,
+      "declaration.goodsShipment.consignment.departureTransportMeans.identificationTypeCode" -> meansOfTransportOnDepartureType,
+      "declaration.goodsShipment.consignment.departureTransportMeans.id" -> meansOfTransportOnDepartureIDNumber
+        .getOrElse(""),
+      "declaration.borderTransportMeans.identificationTypeCode" -> meansOfTransportCrossingTheBorderType,
+      "declaration.borderTransportMeans.id" -> meansOfTransportCrossingTheBorderIDNumber.getOrElse(""),
+      "declaration.borderTransportMeans.registrationNationalityCode" ->
+        allCountries
+          .find(country => meansOfTransportCrossingTheBorderNationality.contains(country.countryName))
+          .map(_.countryCode)
+          .getOrElse(""),
+      "declaration.goodsShipment.consignment.containerCode" -> (if (container) "1" else "0")
+    )
+}
+
+object TransportInformation {
+  implicit val format = Json.format[TransportInformation]
+
+  val id = "TransportInformation"
+
+  def form(): Form[TransportInformation] = Form(mapping)
+
+  import ModeOfTransportCodes._
+  private val allowedModeOfTransportCodes =
```

Comment:
This allowedModeOfTransportCodes you have in object TransportCodes

Reply:
this class will be removed later